### PR TITLE
test42.d: remove Phobos dependency

### PR DESCRIPTION
**Pull request by WalterBright**
dmd tests should not depend on Phobos.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: test/runnable/test42.d [hunk 1]
```diff
@@ -52,7 +50,7 @@ void test2()
 void test3()
 {
     auto i = mixin("__LINE__");
-    writefln("%d", i);
+    printf("%d\n", i);
     assert(i == 54);
```
Comment (thewilsonator) [inline test/runnable/test42.d]: This has changed

Comment (thewilsonator) [reply test/runnable/test42.d]: Please merge once this is fixed.

Comment (WalterBright) [reply test/runnable/test42.d]: ??

Comment (WalterBright) [reply test/runnable/test42.d]: I don't get it. What's wrong?

Comment (thewilsonator) [reply test/runnable/test42.d]:
```
auto i = mixin("__LINE__"); // line 52 therefore i == 52
...
assert(i == 54);
```

Comment (WalterBright) [reply test/runnable/test42.d]: Ah, silly me. Thanks!

Comment (thewilsonator) [reply test/runnable/test42.d]: Im sure you would have found it eventually.

**Pull request merged**